core/unit_test/CMakeLists.txt

```diff
@@ -351,6 +351,13 @@ if(Kokkos_ENABLE_OPENMPTARGET)
     ${CMAKE_CURRENT_BINARY_DIR}/openmptarget/TestOpenMPTarget_Other.cpp
     ${CMAKE_CURRENT_BINARY_DIR}/openmptarget/TestOpenMPTarget_TeamReductionScan.cpp
     ${CMAKE_CURRENT_BINARY_DIR}/openmptarget/TestOpenMPTarget_WorkGraph.cpp
+
+#FIXME_OPENMPTARGET - These unit tests lead to ICE with llvm-16
```

Comment:
Is this due to the refactor

Reply:
No this is not due to the refactor. This is because llvm-16 (against which I was testing) crashes on these unit tests. I did not know how to specialize the loop for Clang/16 in CMAKE . llvm-15 which is used in our CI passes these tests.

Reply:
You should tackle this else where, not as part of a larger refactor.
The description of the PR states that you are just moving things around.

Reply:
Rolled back the changes.

Reply:
> > Is there a reason why you decided to split the files in a different way than what's done in `Serial`, `HIP`, and `Cuda`?
> 
> The current implementation follows the `Cuda` way, the proposed way in this PR follows the style followed by the `OpenACC` backend. I found it more intuitive since each construct gets its own implementation file.

To help have a  software/code organization that's maintainable and that can be associated with other backends:  I think  each (OpenMP)  _directive_ gets its own implementation file, right? Not each OpenMP _construct_? 

I don't know the OpenACC organization  in Kokkos (and this would be a very good comparison point by the way given recent developments I have seen with it) but citing the [OpenMP 5.2 Specification](https://www.openmp.org/wp-content/uploads/OpenMPRefCard-5-2-web.pdf
) as guidance: 'An OpenMP executable directive applies to the succeeding structured block or an OpenMP construct. Each directive starts with #pragma omp. ... A construct consists of a directive and, often, additional base language code. C/C++ C directives are formed exclusively with pragmas. C++ .'  


(Also, this follows standard Computer Science terminology, a construct can be any syntax that provides functionality like a conditional or loop. So, 'if' is a lexical token while 'if(true){}' would be a construct. Also, the '{...}' is a construct while '{' is a lexical token. )